services/proposals/src/propose.py

```diff
@@ -23,30 +24,41 @@ def propose(db_insert_fn: Callable[[Mapping[T, T], T], None]) -> None:
     start_time = time.time()
     client = MongoClient(os.environ["DBCONNECT"])
     logging.info(f'Connected to client: {client}. Setting watch on raw_pdfs collection')
+    pod_name = os.environ["POD_NAME"]
+    podre = re.search(r'\d+$', pod_name)
+    pod_id = int(podre.group()) if podre else None
+    if pod_id is None:
+        logging.error('Error parsing for pod id')
+        return
+    replica_count = int(os.environ["REPLICA_COUNT"])
     db = client.pdfs
     # Open a cursor that will watch for inserts on raw_pdfs
     try:
         with db.raw_pdfs.watch([{'$match': {'operationType': 'insert'}}]) as stream:
             for doc in stream:
-                logging.info('Document found and added to queue')
                 full = doc['fullDocument']
+                obj_id = str(full['_id'])
+                doc_num = int(obj_id, 16)
+                if doc_num % replica_count != pod_id:
+                    continue
+                logging.info('Document found and added to queue')
+                if 'page_data' not in full:
+                    full['page_data'] = []
                 page_data = full['page_data']
-                for page in page_data:
-                    bstring = page['bytes']
-                    bytesio = io.BytesIO(bstring)
-                    img = resize_png(bytesio)
-                    # Convert it back to bytes
-                    resize_bytes_stream = io.BytesIO()
-                    img.save(resize_bytes_stream, format='PNG')
-                    resize_bytes = resize_bytes_stream.getvalue()
+                for page in db.pages.find({'pdf_id' : full['_id']}):
```

Comment:
Instead of adding page information/proposals back onto the pdf, I like the idea of keeping this on the page level object in the pages collection. We can update this in another PR.